src/main/java/org/springframework/data/tarantool/core/query/TarantoolObjectRepositoryQueryExecutor.java

```diff
@@ -22,6 +23,13 @@ public TarantoolRepositoryQueryObjectExecutor(final TarantoolOperations operatio
     @Override
     public Object execute(Object[] parameters) {
         final Class<?> returnedType = queryMethod.getResultProcessor().getReturnedType().getReturnedType();
+        if (operations.getMappingContext().hasPersistentEntityFor(returnedType)) {
+            String spaceName = operations.getMappingContext().getRequiredPersistentEntity(returnedType).getSpaceName();
+            if (queryMethod.isCollectionQuery()) {
```

Comment:
Looks like you can check if the space name is null here and do not repeat the `callForObject`/`callForObjectList` calls twice.

BTW, what does the `hasPersistentEntityFor` do?

Reply:
We can't take `spaceName` if our `returnedType` is not `PersistentEntity` (example `Integer`). Because `operations.getMappingContext().getRequiredPersistentEntity(returnedType)` AFAIK will return null. `hasPersistentEntityFor` just checks if returnType is an entity